pkg/tsdb/cloudwatch/query_transformer.go

```diff
@@ -104,10 +99,119 @@ func (e *cloudWatchExecutor) transformQueryResponsesToQueryResult(cloudwatchResp
 			queryResult.ErrorString = "Cloudwatch GetMetricData error: Too many datapoints requested - your search has been limited. Please try to reduce the time range"
 		}
 
+		eq, err := json.Marshal(executedQueries)
+		if err != nil {
+			plog.Error("Could not marshal executedString struct", err)
+		}
+
+		link, err := buildDeepLink(refID, requestQueries, executedQueries, startTime, endTime)
+		if err != nil {
+			plog.Error("Could not build deep link", err)
+		}
+
+		for _, frame := range frames {
+			frame.Meta = &data.FrameMeta{
+				ExecutedQueryString: string(eq),
+			}
+
+			if link != "" {
+				for _, field := range frame.Fields {
+					field.Config = &data.FieldConfig{
+						Links: []data.DataLink{
+							{
+								Title:       "View in CloudWatch console",
+								TargetBlank: true,
+								URL:         link,
+							},
+						},
+					}
+				}
+			}
+		}
+
 		queryResult.Dataframes = tsdb.NewDecodedDataFrames(frames)
-		queryResult.Meta.Set("gmdMeta", queryMeta)
 		results[refID] = queryResult
 	}
 
 	return results
 }
+
+// Generates a deep link from Grafana to the CloudWatch console. The link params are based on metric(s) for a given query row in the Query Editor
+func buildDeepLink(refID string, requestQueries []*requestQuery, executedQueries []executedQuery, startTime time.Time, endTime time.Time) (string, error) {
+	if isMathExpression(executedQueries) {
+		return "", nil
+	}
+
+	requestQuery := &requestQuery{}
+	for _, rq := range requestQueries {
+		if rq.RefId == refID {
+			requestQuery = rq
+			break
+		}
+	}
+
+	metricItems := []interface{}{}
+	cloudWatchLinkProps := &cloudWatchLink{
+		Title:   refID,
+		View:    "timeSeries",
+		Stacked: false,
+		Region:  requestQuery.Region,
+		Start:   startTime.UTC().Format(time.RFC3339),
+		End:     endTime.UTC().Format(time.RFC3339),
+	}
+
+	expressions := []interface{}{}
+	for _, meta := range executedQueries {
+		if strings.Contains(meta.Expression, "SEARCH(") {
+			expressions = append(expressions, &metricExpression{Expression: meta.Expression})
+		}
+	}
+
+	if len(expressions) != 0 {
+		cloudWatchLinkProps.Metrics = expressions
+	} else {
+		for _, stat := range requestQuery.Statistics {
+			metricStat := []interface{}{requestQuery.Namespace, requestQuery.MetricName}
+			for dimensionKey, dimensionValues := range requestQuery.Dimensions {
+				metricStat = append(metricStat, dimensionKey, dimensionValues[0])
+			}
+			metricStat = append(metricStat, &metricStatMeta{
+				Stat:   *stat,
+				Period: requestQuery.Period,
+			})
+			metricItems = append(metricItems, metricStat)
+		}
+		cloudWatchLinkProps.Metrics = metricItems
+	}
+
+	linkProps, err := json.Marshal(cloudWatchLinkProps)
+	if err != nil {
+		return "", fmt.Errorf("could not marshal link")
```

Comment:
Shouldn't drop the original error:
```suggestion
		return "", fmt.Errorf("could not marshal link: %w", err)
```